webapp/packages/core-ui/src/Hints/useHints.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { action, computed, observable } from 'mobx';
+import { useEffect, useRef } from 'react';
+import type { MenuStateReturn } from 'reakit';
+
+import { useObservableRef } from '@cloudbeaver/core-blocks';
+import type { SqlResultColumn } from '@cloudbeaver/core-sdk';
+import { type IDatabaseDataModel, type IDatabaseResultSet, ResultSetViewAction } from '@cloudbeaver/plugin-data-viewer';
+
+export interface Hint {
+  key: string;
+  value: string;
+  icon?: string | React.ReactElement;
+  title?: string;
+}
+
+export interface HintsState {
+  inputValue: string;
+  changed: boolean;
+  indexToSearchFrom: number;
+  inputRef: React.RefObject<HTMLInputElement> | null;
+  menuRef: React.RefObject<HTMLDivElement>;
+  menu: MenuStateReturn;
+  updateInputValue: (newValue: string) => void;
+  readonly filteredHints: Hint[];
+  separatorLastIndex: number;
+  toggleHints: () => void;
+  getChangedValue: (hint: string) => string;
+}
+
+export function useHints(
+  menu: MenuStateReturn,
+  hints?: Hint[],
+  model?: IDatabaseDataModel<any, IDatabaseResultSet>,
```

Comment:
Hook accepts model, it would better to move it closer to the model ( for example plugin-data-spreadsheet) or change the hook so it would be more universal